src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -190,7 +190,8 @@ public SolutionState WithNewWorkspace(Workspace workspace, int workspaceVersion)
         /// </summary>
         public IReadOnlyList<ProjectId> ProjectIds { get; }
 
-        // [Conditional("DEBUG")]
+        // Only run this in debug builds; the validations (especially the SetEquals calls) can be expensive in large solutions and appear in traces.
```

Comment:
I'm a bit confused. SetEquals should already run in DEBUG only.

Reply:
Ah yes, right, it was the .Any() call.